library/big-float/impl-default.lisp

```diff
@@ -760,8 +761,13 @@ returns the nth SeriesSplit, return the series evaluated to the Nth element."
         ((Tuple x (BFConst f)) (pow x (f)))
         (_ (exp (* y (ln x))))))
     (define (log b x)
-      (/ (ln x) (ln b))))
+      (/ (ln x) (ln b)))
+    (define ee (BFConst (exp 1))))
 
+  (define (bf-ee)
+    "Return the value of ee = exp(1) to the currently set precision."
+    (BFConst (exp 1)))
+  
```

Comment:
I think these are still incorrect. I believe it should be:

```
(declare bf-ee (Unit -> Big-Float))
(define (bf-ee)
  (exp 1))

...
(define ee (BFConst bf-ee))
```

Is my understanding correct that this is the right approach?

Reply:
The issue I'm running into is that `ee` is a member of Exponentiable, in line with this PR. This causes issues if I define `(bf-ee)` before defining the instance:

```
(define (bf-ee)
  (exp 1)) ;; exp isn't defined yet
(coalton-toplevel
   (define-instance Exponentiable Big-Float
    (define ...
    (define ...
    (define (exp ...
    (define ee
```
or after:
```
(coalton-toplevel
   (define-instance Exponentiable Big-Float
    (define ...
    (define ...
    (define (exp ...
    (define ee (BFConst bf-ee)))) ;; bf-ee isn't defined-yet

(define (bf-ee)
  (exp 1)) 
```

Reply:
In that case, `(BFConst (fn (_) (exp 1)))`